packages/gatsby/src/schema/resolvers.ts

```diff
@@ -253,30 +253,29 @@ export function link<TSource, TArgs>(
       }
     }
 
-    const equals = (value: string): any => {
-      return { eq: value }
-    }
-    const oneOf = (value: string): any => {
-      return { in: value }
-    }
-
     // Return early if fieldValue is [] since { in: [] } doesn't make sense
     if (Array.isArray(fieldValue) && fieldValue.length === 0) {
       return fieldValue
     }
 
-    const operator = Array.isArray(fieldValue) ? oneOf : equals
-    const runQueryArgs = args as TArgs & { filter: any }
-    runQueryArgs.filter = options.by
-      .split(`.`)
-      .reduceRight((acc, key, i, { length }) => {
-        return {
-          [key]: i === length - 1 ? operator(acc) : acc,
-        }
-      }, fieldValue)
+    const runQueryArgs = args as any
```

Comment:
Unhappy about this but it's not really changing the status quo. This object is very tricky to type properly due to its semi-dynamic-semi-fixed nature.

Reply:
What's wrong with the previous typing:

```
    const runQueryArgs = args as TArgs & { filter: any }
```

It doesn't give any errors to me with this PR:

Reply:
Not sure at this point. Maybe it was showing errors in an intermediate state but not anymore. I'll revert if so.

Can you help out with the tests?